[DX-1348] Add Inclusive Naming Initiative To Developer Support section menu (#4647)
TykTechnologies/tyk-docs: tyk-docs/content/developer-support/documentation-projects/inclusive-naming.md

- **Create a more welcoming environment**: By using inclusive language, we create a space where everyone feels valued and respected.
- **Enhance accessibility**: Clear and inclusive documentation improves accessibility for all users, regardless of their background or identity.

---

## Phase 1 Tier 1 compliance

### Tier 1 word list

INI sorts terms into word lists, considering both the severity of the term and the level of scrutiny it has received. [INI Tier 1 words](https://inclusivenaming.org/word-lists/tier-1) are considered critical and are recommended to be replaced immediately.

INI have identified that terms included in this list have one or all of the following attributes:

- Strong social consensus within the software development community on replacements
- Are identified by the INI as high-severity terms in need of immediate replacement
- Terms where the impact of change or removal is low: for example, there is little entanglement in low-level systems or standardised language set by standards bodies
- Have passed through all the review stages in Tiers 2 and 3

---

> ### Phase 1 current status

Reviewer: ```suggestion
### Phase 1 - current status
```

   - These words are sourced from Tyk products, in addition to third-party libraries and tooling, e.g. Redis.
   
## Phase 2: Content update

In June 2024, based on the review we then started to update content where possible:

1.  **Explanatory content with INI tier 1 words**: Content on these pages has been rephrased and the work is now completed. 
2.  **Configuration parameters containing INI tier 1 words**: These words are still in our docs, however, we minimised their usage and rephrased their explanatory content to use inclusive words.

## Phase 2: Status

   - **Tyk components**: These words are still in our docs, however, Tyk aims to gradually replace them (in a backwards-compatible way) and update the docs accordingly.
   - **Third-party and dependencies**: There's nothing much we can do at the moment except wait for them to replace these parameters, however, we are committed to updating our docs once this gets done.

The following sections highlight the existing *INI tier 1 words* in our docs, per Tyk component:

### Tyk Gateway

#### Config parameters

> - [allow_master_keys]({{< ref "tyk-dashboard/configuration#enable_master_keys" >}})

Reviewer: ```suggestion
- [allow_master_keys]({{< ref "tyk-dashboard/configuration#allow_master_keys" >}})
```

Reviewer: @dcs3spp fyi, it's from codium

- Terms where the impact of change or removal is low: for example, there is little entanglement in low-level systems or standardized language set by standards bodies
- Have passed through all the review stages in Tiers 2 and 3

---

## Phase #1: Expert review

An initial review of the Tyk documentation was conducted in April 2024 to check which tier 1 words can be replaced, which can't, and why.

### Findings
The main findings of the review are:

1. **Explanatory content with INI tier 1 words**: The content on these pages can be easily rephrased and is now completed.
2. **Configuration parameters containing INI tier 1 words**:
   - Some *INI tier 1 words* are embedded in the code as part of the name of parameters, fields, and keywords.
   - These words are sourced from Tyk products, and third-party libraries, and tooling, e.g. Redis.
   - Possible action:
     - Being part of the source code, we can't simply rephrase and use a different terminology. Such change requires following a deprecation process to ensure there are no breaking changes for our users.
     - For now, we can, however, minimize their usage and rephrase the explanatory content to use inclusive words.
  
   

> ## Phase #2: Content update

Reviewer: ```suggestion
## Phase #2: Removing Tier 1 words from Tyk documentation
```

     - For now, we can minimize their usage and rephrase the explanatory content to use inclusive words.

   
## Phase #2: Removing Tier 1 words from Tyk documentation

In June 2024, based on the review we executed the planned changes to the content in our [documentation](https://github.com/tykTechnologies/tyk-docs/).

### Status update
This is the update on the status of our documentation
1. **Regular explanatory content with INI tier 1 words**: Content in the documentation has been rephrased and the work is now completed. 
2. **Configuration parameters containing INI tier 1 words**: These words are still in our docs, however, we minimized their usage and rephrased their explanatory content to use inclusive words.
   - **Tyk products**: These words are still in our docs, however, Tyk aims to gradually replace them (in a backward-compatible way) and update the docs accordingly.
   - **Third-party and dependencies**: There's nothing much we can do at the moment except wait for them to replace these parameters, however, we are committed to updating our docs once this gets done.

> ### List of Configuration parameters containing INI tier 1 word

Reviewer: ```suggestion
### List of configuration parameters containing INI tier 1 word
```

---
title: "Inclusive Naming Initative"
date: 2024-05-17T15:51:00+01:00
tags: [ "Inclusive Naming Initiative", "Inclusivity", "Inclusive" ]
description: "Explains the inclusive naming initative in relation to Tyk docs"
---

> We are excited to announce the launch of our *Inclusive Naming* project, dedicated to updating our documentation to align with Tier 1 words of the [Inclusive Naming Initiative (INI)](https://inclusivenaming.org/word-lists/tier-1/). This initiative reflects our ongoing commitment to fostering an inclusive and respectful environment within our company and for our users.

Reviewer: Do we need to add a date? once we update the page, the date at the top will change and readers won't know when we announce it

Author: updated with date added

The [Inclusive Naming Initiative](https://inclusivenaming.org/) is a community-driven effort to promote and standardise the use of inclusive language in software and documentation. By adhering to their guidelines, we aim to eliminate terms that can be considered exclusionary, offensive or insensitive and replace them with language that is respectful and welcoming to all.

---

## Purpose

Our commitment to diversity, equity and inclusion is foundational to our values. By updating our documentation to comply with Inclusive Naming Initiative (INI) Tier 1 [words](https://inclusivenaming.org/word-lists/tier-1), we are taking a significant step towards ensuring that our language reflects our dedication to inclusivity. This project will help us:

- **Create a more welcoming environment**: By using inclusive language, we create a space where everyone feels valued and respected.
- **Enhance accessibility**: Clear and inclusive documentation improves accessibility for all users, regardless of their background or identity.

---

## Phase 1 - Removal of Tier 1 Words 

### What is Tier 1 

> The inclusive naming initiative sorts terms into word lists, considering both the severity of the term and the level of scrutiny it has received. Tier 1 [words](https://inclusivenaming.org/word-lists/tier-1) are considered critical and are recommended to be replaced immediately.

Reviewer: ```suggestion
The inclusive naming initiative sorts terms into word lists, considering both the severity of the term and the level of scrutiny it has received. [Tier 1 words](https://inclusivenaming.org/word-lists/tier-1) are considered critical and are recommended to be replaced immediately.
```

---
title: "Inclusive Naming Initative"
date: 2024-05-17T15:51:00+01:00
tags: [ "Inclusive Naming Initiative", "Inclusivity", "Inclusive" ]
description: "Explains the inclusive naming initative in relation to Tyk docs"
---

> We are announcing the *Inclusive Naming* project aimed at updating our documentation to comply with Tier 1 words of the [Inclusive Naming Initiative (INI)](https://inclusivenaming.org/). This initiative is part of our ongoing commitment to foster an inclusive and respectful environment within our company and for our users.

Reviewer: ```suggestion
We are excited to announce the launch of our *Inclusive Naming* project, dedicated to updating our documentation to align with Tier 1 words of the [Inclusive Naming Initiative (INI)](https://inclusivenaming.org/). This initiative reflects our ongoing commitment to fostering an inclusive and respectful environment within our company and for our users.
```

> The [Inclusive Naming Initiative](https://inclusivenaming.org/) is a community-driven effort to promote and standardise the use of inclusive language in software and documentation. By adhering to the guidelines, we aim to eliminate terms that can be considered exclusionary, offensive or insensitive and replace them with language that is respectful and welcoming to all.

Reviewer: Can we add another link for this inclusivenaming.org domain,  instead of having two identical links

Author: updated first instance to link to tier 1 word list

> The [Inclusive Naming Initiative](https://inclusivenaming.org/) is a community-driven effort to promote and standardise the use of inclusive language in software and documentation. By adhering to the guidelines, we aim to eliminate terms that can be considered exclusionary, offensive or insensitive and replace them with language that is respectful and welcoming to all.

Reviewer: ```suggestion
The [Inclusive Naming Initiative](https://inclusivenaming.org/) is a community-driven effort to promote and standardise the use of inclusive language in software and documentation. By adhering to their guidelines, we aim to eliminate terms that can be considered exclusionary, offensive or insensitive and replace them with language that is respectful and welcoming to all.
```

---

## Purpose

Our commitment to diversity, equity and inclusion is foundational to our values. By updating our documentation to comply with Inclusive Naming Initiative (INI) Tier 1 [words](https://inclusivenaming.org/word-lists/tier-1), we are taking a significant step towards ensuring that our language reflects our dedication to inclusivity. This project will help us:

- **Create a more welcoming environment**: By using inclusive language, we create a space where everyone feels valued and respected.
- **Enhance accessibility**: Clear and inclusive documentation improves accessibility for all users, regardless of their background or identity.

---

> ## Tier 1 Words

Reviewer: ```suggestion
## Phase #1 - Removal of Tier 1 Words 
```

The inclusive naming initiative sorts terms into word lists, considering both the severity of the term and the level of scrutiny it has received. Tier 1 [words](https://inclusivenaming.org/word-lists/tier-1) are considered critical and are recommended to be replaced immediately.

Terms included in this list have one or all of the following:

- Strong social consensus within the software development community on replacements
- Are identified by the Inclusive Naming Initiative as high-severity terms in need of immediate replacement
- Terms where the impact of change or removal is low: for example, there is little entanglement in low-level systems or standardised language set by standards bodies
- Have passed through all the review stages in Tiers 2 and 3

At the time of writing the following are considered as Tier 1 keywords:

> - [abort](https://inclusivenaming.org/word-lists/tier-1/abort/)

Author: thanks @letzya I have removed the list of words

- [blackhat-whitehat](https://inclusivenaming.org/word-lists/tier-1/blackhat-whitehat/)
- [cripple](https://inclusivenaming.org/word-lists/tier-1/_cripple/)
- [master](https://inclusivenaming.org/word-lists/tier-1/_master/)
- [master-slave](https://inclusivenaming.org/word-lists/tier-1/_master-slave/)
- [tribe](https://inclusivenaming.org/word-lists/tier-1/tribe/)
- [whitelist](https://inclusivenaming.org/word-lists/tier-1/whitelist/)

---

> ## Tier 1 Occurrences

Reviewer: ```suggestion
## # Tier 1 Occurrences
```

An initial review of the Tyk documentation has been conducted and there are instances of tier 1 keywords for
*Master/Slave* and *Whitelist/Blacklist*.
 

> ### Master and Slave

Reviewer: ```suggestion
#### Master and Slave
```

> There are instances where *master* and *slave* keywords are used within configuration parameters for the following Tyk components:

Reviewer: Did you mean to say that in the following list, we have Tier 1 words that we cannot remove as it's part of the product and we have to wait for that to be changed first?

Author: thanks @letzya I have refactored the content

- **Tyk MDCB Configuration Parameters**:
    - [analytics_storage.master_name]({{< ref "tyk-multi-data-centre/mdcb-configuration-options#analytics_storagemaster_name" >}})
    - [storage.master_name]({{< ref "tyk-multi-data-centre/mdcb-configuration-options#storagemaster_name" >}})
- **Tyk Pump Configuration**:
    - [analytics_storage_config.master_name]({{< ref "tyk-pump/tyk-pump-configuration/tyk-pump-environment-variables#analytics_storage_configmaster_name" >}})

Additionally, content contains instances of *master* under the following circumstances:
- Links to Tyk Component GitHub repositories with a default branch set as master. 
- Tyk Gateway and Tyk Pump content use Redis terminology for master in relation to key storage and analytics. 
- Tyk Classic Developer Portal provides [documentation]({{< ref "tyk-developer-portal/tyk-portal-classic/keycloak-dcr" >}}) that explains how to integrate Tyk with Keycloak using the [OpenID Connect Dynamic Client Registration](https://tools.ietf.org/html/rfc7591) protocol. The example in the guide uses the Keycloak default *master* realm.
- Tyk Bitnami Helm Charts use a service with a DNS name of *tyk-redis-master.tyk.svc*.

---

> ### Whitelist and Blacklist

Reviewer: ```suggestion
#### Whitelist and Blacklist
```
